Ajout de la tâche delete_project

sambarbarin opened the pull request.

Commit:
Ajout de la tâche delete_project

sambarbarin committed:
Fix ansible module in delete_project.yml line 27

sambarbarin committed:
Add new line at the end of yml file

sambarbarin committed:
add space in commented lines

sambarbarin committed:
Supprimer un projet

sambarbarin committed:
Add indication in README

araimbert commented on .config/molecule/config.yml:
```diff
@@ -6,15 +6,15 @@ driver:
-    tmpfs:
-      - /tmp
-      - /run
+#    volumes:
```
A décommenter

araimbert commented on handlers/main.yml:
```diff
@@ -13,6 +13,13 @@
+  register: restart_result
+  ignore_errors: true
+
+- name: skip restart task if previous task failed
```
C'est pourquoi ça ?

sambarbarin replied:
C'est parce que à un moment donné il essayait de redémarrer le service alors qu'il était supprimé, j'ai rajouté ca pour ignoré l'erreur.

araimbert replied:
ha ok alors me plutôt un `when` quand pour qu'il le fasse pas quand li y a pas besoin mais si ca pete il faut pas ignorer les erreurs

sambarbarin replied:
En fait je viens de refaire des tests et ca fonctionne sans, je pense juste qu'hier avec toutes les modifications que j'ai fait il y eu un moment ou ca à du générer une erreur qui n'intervient plus maintenant. Je viens de le retirer et ca passe toujours bien

araimbert commented on molecule/docker/Dockerfile-ubuntu2004:
```diff
@@ -1,4 +1,4 @@
-FROM ubuntu:20.04
+FROM jgeusebroek/ubuntu2004-molecule
```
Vaut mieux remettre comme c'était là

sambarbarin replied:
Je remet FROM ubuntu:20.04 ?

araimbert replied:
Yes!

araimbert commented on tasks/common/delete_project.yml:
```diff
@@ -0,0 +1,54 @@
+- name: Check if the service exists
+  command: systemctl status "{{ steamengine_project_name }}.service"
+  register: service_status
+  ignore_errors: true
```
Pareil, pourquoi t'ignore l'erreur la ?

sambarbarin replied:
C'est parce que tous les projets n'ont pas de service associé il y'a que les types springboot.

sambarbarin replied:
Ca permet de skip le check service existe par exemple les front n'ont pas de .service

araimbert replied:
Idem ici

araimbert commented on tasks/common/delete_project.yml:
```diff
@@ -0,0 +1,54 @@
+    state: absent
+  with_items:
+    - "{{ steamengine_project_user }}"
+    - "{{ steamengine_app_user }}"
```
Ca serait peut-être bien de rajouter dans le README les variables dont il y a besoin

sambarbarin committed:
Update config.yml

sambarbarin committed:
Add variables to delete project in README

sambarbarin committed:
Change image ubuntu:20.04 in Dockerfile

sambarbarin committed:
Remove ignor_errors =true for delete project

sambarbarin committed:
Update main.yml

sambarbarin committed:
Update delete_project.yml

sambarbarin committed:
Update converge.yml

sambarbarin committed:
Update delete_project.yml

J'ai modifié la façon dont je gere les exeptions au cas ou le service serait inexistant ou pas en état de marche.

araimbert commented on molecule/static/converge.yml:
```diff
@@ -1,6 +1,7 @@
 ---
 - name: Converge
   hosts: instance
+  remote_user: sbarbarin
```
a delete

sambarbarin replied:
ok

sambarbarin committed:
Update converge.yml

remove user

araimbert commented on vars/main.yml:
```diff
@@ -39,3 +39,7 @@ steamengine_project_types:
   - symfony
+
+# ADDITIONAL VARIABLES
+steamengine_delete_project: false
```
Elle est pas utilisée au final cette variable ?

sambarbarin committed:
Update main.yml

sambarbarin committed:
Merge branch 'master' into delete_project
